kernel/classes/ezpersistentobject.php

```diff
@@ -920,7 +920,7 @@ public static function fetchObjectList( $def,
             {
                 $db_params["limit"] = $limit["limit"];
             }
-            else
+            elseif( isset( $limit["length"] ) )
```

Comment:
logic of this change does not make sense, the if is already using this `if ( isset( $limit['limit'] ) )`